src/Helpers/PageHelpers/PageHelper.cs

```diff
@@ -407,7 +408,13 @@ private void CheckRetrieveExternalDataAction(List<IAction> actions)
 
             actions.ForEach(action =>
             {
-                if (string.IsNullOrEmpty(action.Properties.URL))
+                var foundSlug = new PageActionSlug();
+                foreach (var slug in action.Properties.PageActionSlugs.Where(_ => _.Environment.ToLower().Equals(_environment.EnvironmentName.ToS3EnvPrefix().ToLower())))
```

Comment:
foreach loop can be removed and replaced with FirstOrDefault.

var foundSlug = action.Properties.PageActionSlugs.FirstOrDefault(_ => _)